src/tools/interop/idt/capture/platform/android/streams/screen/screen.py

```diff
@@ -0,0 +1,100 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+import os
+from typing import TYPE_CHECKING
+
+from utils.artifact import create_standard_log_name, log
+
+from ..base import AndroidStream
+
+if TYPE_CHECKING:
+    from capture.platform.android import Android
+
+logger = log.get_logger(__file__)
+
+
+class ScreenRecorder(AndroidStream):
+
+    def __init__(self, platform: "Android"):
+        self.screen_artifact = None
+        self.screen_phone_out_path = None
+        self.screen_command = None
+        self.screen_proc = None
+        self.logger = logger
+        self.platform = platform
+        self.screen_check_command = "shell dumpsys deviceidle | grep mScreenOn"
+        self.screen_pull = False
+        self.file_counter = 0
+        self.pull_commands: [str] = []
+        self.manifest_file = os.path.join(platform.artifact_dir, "screen_manifest.txt")
+
+    def check_screen(self) -> bool:
+        screen_cmd_output = self.platform.run_adb_command(
+            self.screen_check_command, capture_output=True)
+        return "true" in screen_cmd_output.get_captured_output()
```

Comment:
Should we check for word boundaries for "true" (e.g. via a regex)? So that something like 'construed' or something odd does not match (even if far fetched for sure).

A comment describing what the format of the output is would probably help understanding what we are searching for.

Reply:
Changed to an explicit check!